js/ui/file_manager/ui.file_manager.adaptivity.js

```diff
@@ -27,15 +28,43 @@ class FileManagerAdaptivityControl extends Widget {
 
         this._drawer = this._createComponent($drawer, Drawer, {
             opened: true,
-            template: this.option("drawerTemplate")
+            template: this._createDrawerTemplate.bind(this)
         });
     }
 
+    _createDrawerTemplate(container) {
+        this.option("drawerTemplate")(container);
+        const leftElement = container;
+        const rightElement = this.$element().find(".dx-drawer-content");
+        const minAvailablePosX = 100;
+        const splitter = this._createComponent("<div>", SplitterControl, {
+            container: this.$element(),
+            leftElement,
+            rightElement,
+            minAvailablePosX,
+            onSplitterChanged: this._onSplitterChanged.bind(this)
+        });
+        splitter.$element().appendTo(container);
+        this._leftElement = leftElement;
+        this._rightElement = rightElement;
+        this._splitter = splitter;
+    }
+
     _render() {
         super._render();
         this._checkAdaptiveState();
     }
 
+    _onSplitterChanged(elementsWidth) {
+        this._updateWidth(elementsWidth.actionValue);
```

Comment:
Do you use the `_updateWidth` method in another place? Do you have reasons to create the method?